mlflow/genai/experimental/databricks_trace_archival.py

```diff
@@ -344,20 +380,43 @@ def _do_enable_databricks_archival(
             f"Creating archival configuration for experiment {experiment_id} in {catalog}.{schema}"
         )
 
-        trace_archive_config = DatabricksTraceServerClient().create_trace_destination(
-            experiment_id=experiment_id,
-            catalog=catalog,
-            schema=schema,
-            table_prefix=table_prefix,
-        )
+        # The backend API is idempotent if the same configuration already exists
+        # and does not recreate existing tables.
+        # It will throw ALREADY_EXISTS error if a different configuration already exists
+        # or if the table schema versions have changed. In this case, we create new tables.
+        try:
+            trace_archive_config = DatabricksTraceServerClient().create_trace_destination(
+                experiment_id=experiment_id,
+                catalog=catalog,
+                schema=schema,
+                table_prefix=table_prefix,
+            )
+        except Exception as e:
+            if e.error_code == "ALREADY_EXISTS":
+                # TODO: replace this with an atomic UPDATE operation when backend supports it
+                _logger.info(
+                    f"Trace archival already enabled for experiment {experiment_id}. "
+                    "Deleting existing configuration and trying again. If enablement fails "
+                    "with the new configuration, the old configuration will NOT be restored."
+                )
+                DatabricksTraceServerClient().delete_trace_destination(experiment_id)
```

Comment:
Instead of making this a default action, can we just update above info message with "If you want to link the experiment to a different configuration, call `set_experiment_storage_location(None)` first and try again" and return?